lib/archethic/transaction_chain.ex

```diff
@@ -553,4 +587,206 @@ defmodule Archethic.TransactionChain do
       fn _ -> :ok end
     )
   end
+
+  @doc """
+  Fetch transaction remotely
+
+  If the transaction exists, then its value is returned in the shape of `{:ok, transaction}`. 
+  If the transaction doesn't exist, `{:error, :transaction_not_exists}` is returned.
+
+  If no nodes are available to answer the request, `{:error, :network_issue}` is returned.
+  """
+  @spec fetch_transaction_remotely(address :: Crypto.versioned_hash(), list(Node.t())) ::
+          {:ok, Transaction.t()}
+          | {:error, :transaction_not_exists}
+          | {:error, :transaction_invalid}
+          | {:error, :network_issue}
+  def fetch_transaction_remotely(_, []), do: {:error, :transaction_not_exists}
+
+  def fetch_transaction_remotely(address, nodes) when is_binary(address) and is_list(nodes) do
+    conflict_resolver = fn results ->
+      # Prioritize transactions results over not found
+      with nil <- Enum.find(results, &match?(%Transaction{}, &1)),
+           nil <- Enum.find(results, &match?(%Error{}, &1)) do
+        %NotFound{}
+      else
+        res ->
+          res
+      end
+    end
+
+    case P2P.quorum_read(
+           nodes,
+           %GetTransaction{address: address},
+           conflict_resolver
+         ) do
+      {:ok, %NotFound{}} ->
+        {:error, :transaction_not_exists}
+
+      {:ok, %Error{}} ->
+        {:error, :transaction_invalid}
+
+      {:ok, tx = %Transaction{}} ->
+        {:ok, tx}
+
+      {:error, :network_issue} ->
+        {:error, :network_issue}
+    end
+  end
+
+  @doc """
+  Stream transaction chain remotely
+  """
+  @spec stream_remotely(
+          address :: Crypto.versioned_hash(),
+          list(Node.t()),
+          paging_state :: nil | binary()
+        ) ::
+          Enumerable.t() | list(Transaction.t())
+  def stream_remotely(address, nodes, paging_state \\ nil)
+  def stream_remotely(_, [], _), do: []
+
+  def stream_remotely(address, nodes, paging_state)
+      when is_binary(address) and is_list(nodes) do
+    Stream.resource(
+      fn -> {address, paging_state, 0} end,
+      fn
+        {:end, _size} ->
+          {:halt, address}
+
+        {address, paging_state, size} ->
+          do_stream_chain(nodes, address, paging_state, size)
+      end,
+      fn _ -> :ok end
+    )
+  end
+
+  defp do_stream_chain(nodes, address, paging_state, size) do
+    case fetch_transaction_chain(nodes, address, paging_state) do
+      {transactions, false, _} ->
+        {[transactions], {:end, size + length(transactions)}}
+
+      {transactions, true, paging_state} ->
+        {[transactions], {address, paging_state, size + length(transactions)}}
+    end
+  end
+
+  defp fetch_transaction_chain(
+         nodes,
+         address,
+         paging_state
+       ) do
+    conflict_resolver = fn results ->
+      results
+      |> Enum.sort(&(&1.more? and length(&1.transactions) > length(&2.transactions)))
```

Comment:
This resolver will not work, for exemple if you have `a={more?: false, length: 6}` and `b={more?: false, length: 5}`
as `a.more? == false` the sorted result will be [b,a] but we want [a,b] as a is the longuest chain.
I would recommand something like this
`Enum.sort(&((&1.more? and !&2.more?) or length(&1.transactions) > length(&2.transactions))`